### ISSUE TITLE
ERROR: In procedure free-handle: Wrong type argument in position 1: #<finalized smob 51f7050>

### ISSUE THREAD

**Issue by amirouche** [spk121/guile-curl#3]
I try to dump HN using guile curl and [guile-json](https://github.com/aconchillo/guile-json) but I stumble upon the following error:

```
Backtrace:
12289204
           0 (apply-smob/1 #<catch-closure 4490f40>)

ERROR: In procedure apply-smob/1:
ERROR: In procedure free-handle: Wrong type argument in position 1: #<finalized smob 51f7050>
```

Have a look a [the script I use to do that](https://gist.github.com/amirouche/138a27bdbef5a672a0135f90ca26ec41). Basically I do many call to guile-curl and at some point I get this error.

**Comment by spk121** [comment]
On Monday, August 15, 2016 1:24 AM, amirouche <notifications@github.com> wrote:


>
>
>I try to dump HN using guile curl and guile-json but I stumble upon the following error:
>Backtrace:
12289204 0 (apply-smob/1 #<catch-closure 4490f40>) ERROR: In procedure apply-smob/1:
ERROR: In procedure free-handle: Wrong type argument in position 1: #<finalized smob 51f7050> 
>Have a look a the script I use to do that. Basically I do many call to guile-curl and at some point I get this error.
>—
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly, view it on GitHub, or mute the thread.
>

>

Hmm.  Which version of Guile?

**Comment by amirouche** [comment]
Guile 2.1.3

**Comment by spk121** [comment]
Fixed in 2740cbabd4ee83bbe8eed70dd07e54aecb67adda
